src/arrayMethodJoin.js

```diff
@@ -5,7 +5,47 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joined = '';
+    const joinedArr = [...this];
+    let newSep = separator;
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    switch (newSep) {
+      case '' :
+        for (const el of this) {
+          joined += String(el);
+        }
+
+        return joined;
+
+      case undefined :
+        newSep = ',';
+    }
```

Comment:
I think this switch case is unnecessary, just add default value to separator argument